sopel/tools/jobs.py

```diff
@@ -127,75 +199,284 @@ def _call(self, job):
 class Job(object):
     """Holds information about when a function should be called next.
 
-    :param int interval: number of seconds between calls to ``func``
-    :param func: function to be called
-    :type func: :term:`function`
+    :param intervals: set of intervals; each is a number of seconds between
+                      calls to ``handler``
+    :type intervals: :term:`iterable`
+    :param str plugin: optional job's plugin
+    :param str label: optional job's label
+    :param handler: function to be called when the job is ready to execute
+    :type handler: :term:`function`
+    :param str doc: option job's documentation
 
     Job is a simple structure that holds information about when a function
-    should be called next. They can be put in a priority queue, in which case
-    the Job that should be executed next is returned.
+    should be called next. They are best used with a :class:`Scheduler`
+    that will manage job execution when they are ready.
 
-    Calling :meth:`next` modifies the Job object with the next time it should
-    execute. Current time is used to decide when the job should be executed
-    next so it should only be called right after the function was called.
-    """
+    The :term:`function` to execute is the ``handler``, which must be a
+    callable with this signature::
+
+        def handler(manager):
+            # perform action periodically
+            # return is optional
+
+    The ``manager`` parameter can be any kind of object; usually it's an
+    instance of :class:`sopel.bot.Sopel`.
+
+    When a job is ready, you can execute it by calling its :meth:`execute`
+    method (providing the appropriate ``manager`` argument)::
+
+        if job.is_ready_to_run(time.time()):
+            job.execute(manager)  # marked as running
+            # "next times" have been updated, the job is not running
```

Comment:
```suggestion
            # "next times" have been updated; the job is not running
```

Literally the hardest punctuation mark to learn how to use. 😹